description: add automatically generated files to .gitignore
author: Michael Gilbert <mgilbert@debian.org>

--- a/configure.ac
+++ b/configure.ac
@@ -3484,6 +3484,9 @@ dnl End of auto-generated output command
 
 dnl Main ignore file
 
+WINE_IGNORE_FILE("configure")
+WINE_IGNORE_FILE("tools/config.*")
+WINE_IGNORE_FILE("include/config.h.in")
 WINE_IGNORE_FILE(".gitignore")
 WINE_IGNORE_FILE("Makefile")
 WINE_IGNORE_FILE("TAGS")
--- a/Makefile.in
+++ b/Makefile.in
@@ -36,7 +36,7 @@ include/stamp-h: $(srcdir)/include/confi
 
 clean:: __clean__
 distclean:: clean
-	$(RM) config.* configure.lineno TAGS tags include/config.h include/stamp-h Makefile Make.tmp .gitignore
+	$(RM) config.* configure.lineno TAGS tags include/config.h include/stamp-h Makefile Make.tmp
 	$(RM) -r autom4te.cache
 
 # Rules for uninstalling
