Description: Fix a typo in the msiexec manpage.
Author: Jens Reyer <jre.winesim@gmail.com>
Forwarded: http://source.winehq.org/patches/data/121939

--- a/programs/msiexec/msiexec.man.in
+++ b/programs/msiexec/msiexec.man.in
@@ -76,7 +76,7 @@ Log all error messages.
 .IP "\fBa\fR"
 Log start of actions.
 .IP "\fBr\fR"
-Log action specifc records.
+Log action specific records.
 .IP "\fBu\fR"
 Log user requests.
 .IP "\fBc\fR"
