description: fix macro not defined errors from man
author: Jens Reyer <jre.winesim@gmail.com>

--- a/loader/wine.pl.UTF-8.man.in
+++ b/loader/wine.pl.UTF-8.man.in
@@ -263,8 +263,8 @@ symboliczene dowiązanie w formie $WINEP
 Napędy DOS również są określone przez dowiązania symboliczne; Dla przykładu jeżeli napęd D:
 odpowiadałby napędowi CDROM zamontowanemu w /mnt/cdrom, miałbyś dowiązanie symboliczne
 $WINEPREFIX/dosdevices/d: -> /mnt/cdrom. Urządzenia Uniksowe odpowiadające
-napędom DOS mogą być określone w ten sam sposób, z użyciem '::' zamiast
-':'. Tak więc dla poprzedniego przykładu, jeżeli urządzenie CDROM byłoby zamontowane
+napędom DOS mogą być określone w ten sam sposób, z użyciem '::' zamiast ':'.
+Tak więc dla poprzedniego przykładu, jeżeli urządzenie CDROM byłoby zamontowane
 z /dev/hdc, to odpowiadające dowiązanie symboliczne wyglądałoby następująco
 $WINEPREFIX/dosdevices/d:: -> /dev/hdc.
 .SH AUTORZY
--- a/tools/winemaker/winemaker.de.UTF-8.man.in
+++ b/tools/winemaker/winemaker.de.UTF-8.man.in
@@ -58,7 +58,7 @@ im Includepfad suchen und die entspreche
 Groß-/Kleinschreibweise austauschen.
 .PP
 - Winemaker wird ebenso andere, exotischere Probleme wie die Benutzung von
-'#pragma pack', 'afxres.h' in nicht-MFC-Projekten und mehr untersuchen.
+\&'#pragma pack', 'afxres.h' in nicht-MFC-Projekten und mehr untersuchen.
 Sollte etwas ungwöhnliches vorkommen, wird Winemaker Sie warnen.
 .PP
 - Winemaker kann eine ganze Verzeichnisstruktur auf einmal durchsuchen,
@@ -198,9 +198,9 @@ $ winemaker --lower-uppercase -DSTRICT .
 Damit scannt Winemaker das aktuelle Verzeichnis und die Unterverzeichnisse nach
 Quellcodedateien. Jede Datei und jedes Verzeichnis, das ganz in Großbuchstaben
 geschrieben ist, wird in Kleinbuchstaben umbenannt. Danach werden alle Quellcodedateien
-an die Kompilierung mit Winelib angepasst und Makefiles erzeugt.
-'-DSTRICT' gibt an, dass das STRICT-Makro gesetzt sein muss, um diesen Quellcode
-zu kompilieren. Letztendlich wird Winemaker die globale Makefile erzeugen.
+an die Kompilierung mit Winelib angepasst und Makefiles erzeugt. '-DSTRICT' gibt
+an, dass das STRICT-Makro gesetzt sein muss, um diesen Quellcode zu kompilieren.
+Letztendlich wird Winemaker die globale Makefile erzeugen.
 .PP
 Der nächste Schritt wäre dann:
 .PP
