description: disable external downloading
author: Michael Gilbert <mgilbert@debian.org>

--- a/dlls/appwiz.cpl/addons.c
+++ b/dlls/appwiz.cpl/addons.c
@@ -731,11 +731,8 @@ BOOL install_addon(addon_t addon_type)
      */
     if (install_from_registered_dir() == INSTALL_NEXT
         && install_from_default_dir() == INSTALL_NEXT
-        && install_from_cache() == INSTALL_NEXT
-        && (url = get_url()))
+        && install_from_cache() == INSTALL_NEXT)
         DialogBoxW(hInst, addon->dialog_template, 0, installer_proc);
 
-    heap_free(url);
-    url = NULL;
     return TRUE;
 }
