description: fix pointer cast size mismatch

--- a/dlls/winmm/winmm.c
+++ b/dlls/winmm/winmm.c
@@ -41,6 +41,7 @@
 
 #include <stdio.h>
 #include <stdarg.h>
+#include <stdint.h>
 #include <string.h>
 
 #define NONAMELESSUNION
@@ -930,6 +931,8 @@ typedef struct WINE_MIDIStream {
  */
 static	BOOL	MMSYSTEM_GetMidiStream(HMIDISTRM hMidiStrm, WINE_MIDIStream** lpMidiStrm, WINE_MIDI** lplpwm)
 {
+    intptr_t temp;
+
     WINE_MIDI* lpwm = (LPWINE_MIDI)MMDRV_Get(hMidiStrm, MMDRV_MIDIOUT, FALSE);
 
     if (lplpwm)
@@ -939,7 +942,8 @@ static	BOOL	MMSYSTEM_GetMidiStream(HMIDI
 	return FALSE;
     }
 
-    *lpMidiStrm = (WINE_MIDIStream*)lpwm->mod.rgIds.dwStreamID;
+    temp = lpwm->mod.rgIds.dwStreamID;
+    *lpMidiStrm = (WINE_MIDIStream*)temp;
 
     return *lpMidiStrm != NULL;
 }
