description: make winegcc use -unstable suffix
author: Michael Gilbert <mgilbert@debian.org>
Index: wine/tools/winegcc/winegcc.c
===================================================================
--- wine.orig/tools/winegcc/winegcc.c	2013-09-01 23:21:31.688150546 +0000
+++ wine/tools/winegcc/winegcc.c	2013-09-01 23:21:31.688150546 +0000
@@ -850,7 +850,7 @@
             add_library(opts, lib_dirs, files, "kernel32");
             add_library(opts, lib_dirs, files, "ntdll");
         }
-        if (opts->shared && !opts->nostdlib) add_library(opts, lib_dirs, files, "wine");
+        if (opts->shared && !opts->nostdlib) add_library(opts, lib_dirs, files, "wine-unstable");
         if (!opts->shared && opts->use_msvcrt && opts->target_platform == PLATFORM_CYGWIN)
             add_library(opts, lib_dirs, files, "msvcrt");
 
@@ -936,7 +936,7 @@
         add_library(opts, lib_dirs, files, "kernel32");
         add_library(opts, lib_dirs, files, "ntdll");
     }
-    if (!opts->nostdlib) add_library(opts, lib_dirs, files, "wine");
+    if (!opts->nostdlib) add_library(opts, lib_dirs, files, "wine-unstable");
 
     /* run winebuild to generate the .spec.o file */
     spec_args = get_winebuild_args( opts );
