description: add wine suffix to paths
Index: wine/Make.vars.in
===================================================================
--- wine.orig/Make.vars.in	2013-06-16 19:43:48.000000000 +0000
+++ wine/Make.vars.in	2013-06-16 19:44:02.000000000 +0000
@@ -8,7 +8,7 @@
 datadir         = @datadir@
 infodir         = @infodir@
 mandir          = @mandir@
-fontdir         = $(datadir)/wine/fonts
+fontdir         = $(datadir)/wine$(DATASUFFIX)/fonts
 includedir      = @includedir@/wine
 dlldir          = @libdir@/wine
 fakedlldir      = $(dlldir)/fakedlls
@@ -71,7 +71,7 @@
 WRC             = $(TOOLSDIR)/tools/wrc/wrc$(TOOLSEXT)
 LIBPORT         = $(top_builddir)/libs/port/libwine_port.a
 LIBWPP          = $(top_builddir)/libs/wpp/libwpp.a
-LIBWINE         = -L$(top_builddir)/libs/wine -lwine
+LIBWINE         = -L$(top_builddir)/libs/wine -lwine$(DATASUFFIX)
 WINEGCC         = $(TOOLSDIR)/tools/winegcc/winegcc $(TARGETFLAGS) -B$(TOOLSDIR)/tools/winebuild --sysroot=$(top_builddir) @UNWINDFLAGS@
 CROSSWINEGCC    = $(TOOLSDIR)/tools/winegcc/winegcc $(CROSSTARGET:%=-b %) -B$(TOOLSDIR)/tools/winebuild --sysroot=$(top_builddir) --lib-suffix=.cross.a
 LDPATH          = @LDPATH@
Index: wine/tools/Makefile.in
===================================================================
--- wine.orig/tools/Makefile.in	2013-06-16 19:43:48.000000000 +0000
+++ wine/tools/Makefile.in	2013-06-16 19:44:02.000000000 +0000
@@ -30,7 +30,7 @@
 INSTALLDIRS = \
 	$(DESTDIR)$(bindir) \
 	$(DESTDIR)$(datadir)/applications \
-	$(DESTDIR)$(datadir)/wine \
+	$(DESTDIR)$(datadir)/wine$(DATASUFFIX) \
 	$(DESTDIR)$(mandir)/man$(prog_manext) \
 	$(DESTDIR)$(mandir)/de.UTF-8/man$(prog_manext) \
 	$(DESTDIR)$(mandir)/fr.UTF-8/man$(prog_manext)
@@ -66,8 +66,8 @@
 	sed -e 's,@bindir\@,$(bindir),g' $(srcdir)/wineapploader.in >$@ || ($(RM) $@ && false)
 
 install install-lib:: wine.inf $(INSTALLDIRS)
-	$(INSTALL_DATA) wine.inf $(DESTDIR)$(datadir)/wine/wine.inf
-	$(INSTALL_DATA) $(srcdir)/l_intl.nls $(DESTDIR)$(datadir)/wine/l_intl.nls
+	$(INSTALL_DATA) wine.inf $(DESTDIR)$(datadir)/wine$(DATASUFFIX)/wine.inf
+	$(INSTALL_DATA) $(srcdir)/l_intl.nls $(DESTDIR)$(datadir)/wine$(DATASUFFIX)/l_intl.nls
 	$(INSTALL_DATA) $(srcdir)/wine.desktop $(DESTDIR)$(datadir)/applications/wine.desktop
 	-$(UPDATE_DESKTOP_DATABASE)
 
@@ -79,8 +79,8 @@
 	$(INSTALL_DATA) winemaker.fr.man $(DESTDIR)$(mandir)/fr.UTF-8/man$(prog_manext)/winemaker.$(prog_manext)
 
 uninstall::
-	$(RM) $(DESTDIR)$(datadir)/wine/wine.inf \
-		$(DESTDIR)$(datadir)/wine/l_intl.nls \
+	$(RM) $(DESTDIR)$(datadir)/wine$(DATASUFFIX)/wine.inf \
+		$(DESTDIR)$(datadir)/wine$(DATASUFFIX)/l_intl.nls \
 		$(DESTDIR)$(datadir)/applications/wine.desktop \
 		$(DESTDIR)$(bindir)/winemaker \
 		$(DESTDIR)$(mandir)/de.UTF-8/man$(prog_manext)/winemaker.$(prog_manext) \
Index: wine/tools/winedump/search.c
===================================================================
--- wine.orig/tools/winedump/search.c	2013-06-16 19:43:48.000000000 +0000
+++ wine/tools/winedump/search.c	2013-06-16 19:44:02.000000000 +0000
@@ -98,7 +98,7 @@
       if (VERBOSE)
         puts (grep_buff);
 
-      cmd = str_create (5, "function_grep.pl ", sym->symbol,
+      cmd = str_create (5, DATADIR "function_grep.pl ", sym->symbol,
                         " \"", grep_buff, "\"");
 
       if (VERBOSE)
Index: wine/tools/winedump/Makefile.in
===================================================================
--- wine.orig/tools/winedump/Makefile.in	2013-06-16 19:43:48.000000000 +0000
+++ wine/tools/winedump/Makefile.in	2013-06-16 19:44:02.000000000 +0000
@@ -27,6 +27,8 @@
 
 INSTALLDIRS = $(DESTDIR)$(bindir) $(DESTDIR)$(mandir)/man$(prog_manext)
 
+MODCFLAGS = -DDATADIR=\"$(datadir)/wine$(DATASUFFIX)/\"
+
 all: $(PROGRAMS)
 
 @MAKE_RULES@
@@ -35,6 +37,7 @@
 	$(CC) $(CFLAGS) -o winedump$(EXEEXT) $(OBJS) $(LIBPORT) $(LDFLAGS)
 
 install install-dev:: $(PROGRAMS) $(DESTDIR)$(bindir)
+	mkdir -p $(DESTDIR)/$(datadir)/wine$(DATASUFFIX)
 	$(INSTALL_PROGRAM) winedump$(EXEEXT) $(DESTDIR)$(bindir)/winedump$(EXEEXT)
 	$(INSTALL_SCRIPT) $(srcdir)/function_grep.pl $(DESTDIR)$(bindir)/function_grep.pl
 
Index: wine/dlls/Makedll.rules.in
===================================================================
--- wine.orig/dlls/Makedll.rules.in	2013-06-16 19:43:48.000000000 +0000
+++ wine/dlls/Makedll.rules.in	2013-06-16 19:44:02.000000000 +0000
@@ -17,7 +17,7 @@
 IMPLIBFLAGS = $(TARGETFLAGS) $(EXTRAIMPLIBFLAGS)
 IMPORTLIBFILE = $(IMPORTLIB:%=lib%.@IMPLIBEXT@)
 STATICIMPLIB  = $(IMPORTLIBFILE:.def=.def.a)
-INSTALLDIRS = $(DESTDIR)$(dlldir) $(DESTDIR)$(fakedlldir) $(DESTDIR)$(datadir)/wine
+INSTALLDIRS = $(DESTDIR)$(dlldir) $(DESTDIR)$(fakedlldir) $(DESTDIR)$(datadir)/wine$(DATASUFFIX)
 
 @MAKE_RULES@
 
Index: wine/dlls/wineps.drv/Makefile.in
===================================================================
--- wine.orig/dlls/wineps.drv/Makefile.in	2013-06-16 19:43:48.000000000 +0000
+++ wine/dlls/wineps.drv/Makefile.in	2013-06-16 19:44:02.000000000 +0000
@@ -68,12 +68,12 @@
 
 @MAKE_DLL_RULES@
 
-install-ppd:: $(DESTDIR)$(datadir)/wine dummy
-	$(INSTALL_DATA) $(srcdir)/generic.ppd $(DESTDIR)$(datadir)/wine/generic.ppd
+install-ppd:: $(DESTDIR)$(datadir)/wine$(DATASUFFIX) dummy
+	$(INSTALL_DATA) $(srcdir)/generic.ppd $(DESTDIR)$(datadir)/wine$(DATASUFFIX)/generic.ppd
 
 .PHONY: install-ppd
 
 @WOW64_DISABLE@ install install-lib:: install-ppd
 
 uninstall::
-	$(RM) $(DESTDIR)$(datadir)/wine/generic.ppd
+	$(RM) $(DESTDIR)$(datadir)/wine$(DATASUFFIX)/generic.ppd
Index: wine/configure
===================================================================
--- wine.orig/configure	2013-06-16 19:43:48.000000000 +0000
+++ wine/configure	2013-06-16 19:44:02.000000000 +0000
@@ -6974,19 +6974,19 @@
     esac
 
     LIBWINE_RULES="
-all: libwine.so
+all: libwine\$(DATASUFFIX).so
 
-libwine.so: \$(OBJS) Makefile.in
+libwine\$(DATASUFFIX).so: \$(OBJS) Makefile.in
 	\$(CC) -shared \$(OBJS) \$(EXTRALIBS) \$(LDFLAGS) \$(LIBS) -o \$@
 
-install install-lib:: libwine.so \$(DESTDIR)\$(libdir) dummy
-	\$(INSTALL_PROGRAM) libwine.so \$(DESTDIR)\$(libdir)/libwine.so
+install install-lib:: libwine\$(DATASUFFIX).so \$(DESTDIR)\$(libdir) dummy
+	\$(INSTALL_PROGRAM) libwine\$(DATASUFFIX).so \$(DESTDIR)\$(libdir)/libwine\$(DATASUFFIX).so
 
 uninstall::
-	\$(RM) \$(DESTDIR)\$(libdir)/libwine.so
+	\$(RM) \$(DESTDIR)\$(libdir)/libwine\$(DATASUFFIX).so
 
 clean::
-	\$(RM) libwine.so version.c
+	\$(RM) libwine\$(DATASUFFIX).so version.c
 "
 
     ;;
@@ -7043,7 +7043,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_cflags__fPIC__shared__Wl__soname_confest_so_1" >&5
 $as_echo "$ac_cv_cflags__fPIC__shared__Wl__soname_confest_so_1" >&6; }
 if test $ac_cv_cflags__fPIC__shared__Wl__soname_confest_so_1 = yes; then :
-  LDSHARED="\$(CC) -shared -Wl,-soname,libwine.so.\$(SOVERSION)"
+  LDSHARED="\$(CC) -shared -Wl,-soname,libwine\$(DATASUFFIX).so.\$(SOVERSION)"
 else
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler supports -fPIC -shared -Wl,-h,confest.so.1" >&5
 $as_echo_n "checking whether the compiler supports -fPIC -shared -Wl,-h,confest.so.1... " >&6; }
@@ -7068,7 +7068,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_cflags__fPIC__shared__Wl__h_confest_so_1" >&5
 $as_echo "$ac_cv_cflags__fPIC__shared__Wl__h_confest_so_1" >&6; }
 if test $ac_cv_cflags__fPIC__shared__Wl__h_confest_so_1 = yes; then :
-  LDSHARED="\$(CC) -shared -Wl,-h,libwine.so.\$(SOVERSION)"
+  LDSHARED="\$(CC) -shared -Wl,-h,libwine\$(DATASUFFIX).so.\$(SOVERSION)"
 fi
 fi
 
@@ -7367,34 +7367,34 @@
 $as_echo "$ac_cv_c_dll_unixware" >&6; }
       if test "$ac_cv_c_dll_unixware" = "yes"
       then
-        LDSHARED="\$(CC) -Wl,-G,-h,libwine.so.\$(SOVERSION)"
+        LDSHARED="\$(CC) -Wl,-G,-h,libwine\$(DATASUFFIX).so.\$(SOVERSION)"
         LDDLLFLAGS="-Wl,-G,-B,symbolic"
       fi
     fi
     LIBWINE_RULES="
-all: libwine.so
+all: libwine\$(DATASUFFIX).so
 
-libwine.so.\$(VERSION): \$(OBJS) wine.map Makefile.in
+libwine\$(DATASUFFIX).so.\$(VERSION): \$(OBJS) wine.map Makefile.in
 	$LDSHARED \$(OBJS) \$(EXTRALIBS) \$(LDFLAGS) \$(LIBS) -o \$@
 
-libwine.so.\$(SOVERSION): libwine.so.\$(VERSION)
-	\$(RM) \$@ && \$(LN_S) libwine.so.\$(VERSION) \$@
+libwine\$(DATASUFFIX).so.\$(SOVERSION): libwine\$(DATASUFFIX).so.\$(VERSION)
+	\$(RM) \$@ && \$(LN_S) libwine\$(DATASUFFIX).so.\$(VERSION) \$@
 
-libwine.so: libwine.so.\$(SOVERSION)
-	\$(RM) \$@ && \$(LN_S) libwine.so.\$(SOVERSION) \$@
+libwine\$(DATASUFFIX).so: libwine\$(DATASUFFIX).so.\$(SOVERSION)
+	\$(RM) \$@ && \$(LN_S) libwine\$(DATASUFFIX).so.\$(SOVERSION) \$@
 
-install install-lib:: libwine.so.\$(VERSION) \$(DESTDIR)\$(libdir) dummy
-	\$(INSTALL_PROGRAM) libwine.so.\$(VERSION) \$(DESTDIR)\$(libdir)/libwine.so.\$(VERSION)
-	cd \$(DESTDIR)\$(libdir) && \$(RM) libwine.so.\$(SOVERSION) && \$(LN_S) libwine.so.\$(VERSION) libwine.so.\$(SOVERSION)
+install install-lib:: libwine\$(DATASUFFIX).so.\$(VERSION) \$(DESTDIR)\$(libdir) dummy
+	\$(INSTALL_PROGRAM) libwine\$(DATASUFFIX).so.\$(VERSION) \$(DESTDIR)\$(libdir)/libwine\$(DATASUFFIX).so.\$(VERSION)
+	cd \$(DESTDIR)\$(libdir) && \$(RM) libwine\$(DATASUFFIX).so.\$(SOVERSION) && \$(LN_S) libwine\$(DATASUFFIX).so.\$(VERSION) libwine\$(DATASUFFIX).so.\$(SOVERSION)
 
 install install-dev:: \$(DESTDIR)\$(libdir) dummy
-	cd \$(DESTDIR)\$(libdir) && \$(RM) libwine.so && \$(LN_S) libwine.so.\$(VERSION) libwine.so
+	cd \$(DESTDIR)\$(libdir) && \$(RM) libwine\$(DATASUFFIX).so && \$(LN_S) libwine\$(DATASUFFIX).so.\$(VERSION) libwine\$(DATASUFFIX).so
 
 uninstall::
-	-cd \$(DESTDIR)\$(libdir) && \$(RM) libwine.so libwine.so.\$(VERSION) libwine.so.\$(SOVERSION)
+	-cd \$(DESTDIR)\$(libdir) && \$(RM) libwine\$(DATASUFFIX).so libwine\$(DATASUFFIX).so.\$(VERSION) libwine\$(DATASUFFIX).so.\$(SOVERSION)
 
 clean::
-	\$(RM) libwine.so.\$(SOVERSION) libwine.so.\$(VERSION) version.c
+	\$(RM) libwine\$(DATASUFFIX).so.\$(SOVERSION) libwine\$(DATASUFFIX).so.\$(VERSION) version.c
 "
 
     ;;
Index: wine/libs/wine/Makefile.in
===================================================================
--- wine.orig/libs/wine/Makefile.in	2013-06-16 19:43:48.000000000 +0000
+++ wine/libs/wine/Makefile.in	2013-06-16 19:44:02.000000000 +0000
@@ -100,7 +100,7 @@
 	-DLIB_TO_BINDIR=\"`$(RELPATH) $(libdir) $(bindir)`\" \
 	-DLIB_TO_DLLDIR=\"`$(RELPATH) $(libdir) $(dlldir)`\" \
 	-DBIN_TO_DLLDIR=\"`$(RELPATH) $(bindir) $(dlldir)`\" \
-	-DBIN_TO_DATADIR=\"`$(RELPATH) $(bindir) $(datadir)/wine`\"
+	-DBIN_TO_DATADIR=\"`$(RELPATH) $(bindir) $(datadir)/wine$(DATASUFFIX)`\"
 
 config.o: config.c $(RELPATH)
 	$(CC) -c $(ALLCFLAGS) -o $@ $(srcdir)/config.c $(CONFIGDIRS)
Index: wine/programs/Makeprog.rules.in
===================================================================
--- wine.orig/programs/Makeprog.rules.in	2013-06-16 19:57:59.000000000 +0000
+++ wine/programs/Makeprog.rules.in	2013-06-16 19:58:18.473665901 +0000
@@ -11,7 +11,7 @@
 
 DLLFLAGS    = @DLLFLAGS@
 DEFS        = -D__WINESRC__ $(EXTRADEFS)
-ALL_LIBS    = $(DELAYIMPORTS:%=-l%) $(IMPORTS:%=-l%) $(EXTRALIBS) -lwine $(LIBPORT) $(LDFLAGS) $(LIBS)
+ALL_LIBS    = $(DELAYIMPORTS:%=-l%) $(IMPORTS:%=-l%) $(EXTRALIBS) -lwine$(DATASUFFIX) $(LIBPORT) $(LDFLAGS) $(LIBS)
 INSTALLDIRS = $(DESTDIR)$(bindir) $(DESTDIR)$(dlldir) $(DESTDIR)$(fakedlldir) $(DESTDIR)$(mandir)/man$(prog_manext)
 
 @MAKE_RULES@
